store/dataStore.ts

```diff
@@ -1,33 +1,77 @@
 import { create } from 'zustand';
 import { produce } from 'immer';
-import { IUser } from '@/api/IapiFunctions';
+import {
+  INFLTeam,
+  IUser,
+  IUserWeeklyPick,
+  IWeeklyPicks,
+} from '@/api/IapiFunctions';
 
 interface IDataStoreState {
+  NFLTeam: INFLTeam;
   user: IUser;
+  userWeeklyPick: IUserWeeklyPick;
+  weeklyPicks: IWeeklyPicks;
 }
-
 interface IDataStoreAction {
-  updateUser: (id: IUser['id'], email: IUser['email']) => void;
   resetUser: () => void;
+  updateNFLTeam: (NFLTeam: {
+    teamName: INFLTeam['teamName'];
+    teamLogo: INFLTeam['teamLogo'];
+  }) => void;
+  updateUser: (id: IUser['id'], email: IUser['email']) => void;
+  updateUserWeeklyPick: (
+    id: IUser['id'],
+    weekNumber: IUserWeeklyPick['weekNumber'],
+  ) => void;
+  updateWeeklyPicks: (weeklyPicks: {
+    gameId: IWeeklyPicks['gameId'];
+    gameWeekId: IWeeklyPicks['gameWeekId'];
+    userResults: IWeeklyPicks['userResults'];
+  }) => void;
 }
 
 export interface DataStore extends IDataStoreState, IDataStoreAction {}
 
 const initialState: IDataStoreState = {
+  NFLTeam: {
+    teamName: '',
+    teamLogo: '',
+  },
   user: {
     id: '',
     email: '',
   },
+  userWeeklyPick: {
+    userId: '',
+    weekNumber: '',
+  },
+  weeklyPicks: {
+    gameId: '',
+    gameWeekId: '',
+    userResults: '',
+  },
 };
 
 export const useDataStore = create<DataStore>((set) => ({
   ...initialState,
+  resetUser: () => set({ user: initialState.user }),
+  updateNFLTeam: ({ teamName, teamLogo }) =>
```

Comment:
Q: How come the props here are not `{NFLTeam: {teamName, teamLogo}}`, but the set props are set like that?